hardware/lcd/hd44780_i2c.c

```diff
@@ -97,12 +118,13 @@ output_nibble(uint8_t rs, uint8_t nibble, uint8_t en)
   if (rs)
     lcd_data |= _BV(HD44780_PCF8574x_RS);
 
+#if HD44780_TYPE != HD44780_WDC2704
```

Comment:
Really disable backlight support for every connection type of this display when it is only required for i2c?